src/internal/git/utils.go

```diff
@@ -43,33 +48,33 @@ func MutateGitUrlsInText(host string, text string, gitUser string) string {
 }
 
 func transformURLtoRepoName(url string) (string, error) {
-	// For further explanation: https://regex101.com/library/UfILls and https://regex101.com/rary/UfILls
-	findRegex := regexp.MustCompile(`\/([\w\-]+)(.git)?(@([\w\-\.]+))?$`)
-	substrings := findRegex.FindStringSubmatch(url)
-	if len(substrings) == 0 {
-		// the first element in the return substrings is
+	matches := gitURLRegex.FindStringSubmatch(url)
+	if len(matches) == 0 {
+		// Unable to find a substring match for the regex
 		return "", fmt.Errorf("unable to get extract the repoName from the url %s", url)
 	}
 
-	// NOTE: The first element in the returned substrings is the combination of all the rest of the substrings....
-	//       So just skip the first element so we can get a hash without the version tag
-	repoName := substrings[1]
+	repoName := matches[gitURLRegex.SubexpIndex("repo")]
+	// NOTE: We remove the .git and protocol so that https://zarf.dev/repo.git and http://zarf.dev/repo
+	// resolve to the same repp (as they would in real life)
+	sanitizedURL := matches[gitURLRegex.SubexpIndex("hostPath")] + "/" + repoName + matches[gitURLRegex.SubexpIndex("atRef")]
 
 	// Add sha1 hash of the repoName to the end of the repo
 	hasher := sha1.New()
-	_, _ = io.WriteString(hasher, url)
+	_, _ = io.WriteString(hasher, sanitizedURL)
+
 	sha1Hash := hex.EncodeToString(hasher.Sum(nil))
 	newRepoName := repoName + "-" + sha1Hash
 
 	return newRepoName, nil
 }
 
-func transformURL(baseUrl string, url string, username string) (string, error) {
-	replaced, err := transformURLtoRepoName(url)
+func transformURL(baseURL string, url string, username string) (string, error) {
+	repoName, err := transformURLtoRepoName(url)
 	if err != nil {
 		return "", err
 	}
-	output := baseUrl + "/" + username + "/" + replaced
+	output := baseURL + "/" + username + "/" + repoName
```

Comment:
Not a must change, as there are plenty of legacy offenders in this file already, but we've been trying to move to `fmt.Sprintf()` vs string concatenation whenever we touch older code.

Reply:
done